internal/pipe/build/build.go

```diff
@@ -113,6 +118,79 @@ func runPipeOnBuild(ctx *context.Context, build config.Build) error {
 	return g.Wait()
 }
 
+func proxy(ctx *context.Context, build config.Build) (config.Build, error) {
+	if !build.IsProxied() {
+		return build, nil
+	}
+
+	template := tmpl.New(ctx)
+
+	proxy, err := template.Apply(build.Proxy.Path)
+	if err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	version, err := template.Apply(build.Proxy.Version)
+	if err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	log := log.WithField("id", build.ID)
+	log.Infof("proxying %s@%s", proxy, version)
+
+	template = template.WithExtraFields(tmpl.Fields{
+		"Proxy":   proxy,
```

Comment:
yeah, you can build multiple cmds for example and put in the same archive (or different ones too).

I think the question is: should we allow to build anything that is not the current module? If yes, this setting needs to be somewhere else, otherwise it can be per build (as is on this pr right now), which allows each build to define its own module to be used 🤔

Reply:
Yeh I think this is a key point from the user's perspective. I'm not suggesting/proposing any change in scope of what `goreleaser` does.

So building on my question, how much of the following example falls within the scope of `goreleaser` today? The context is a "release" for the project `github.com/hello/world` that includes binaries for main packages:

1. `github.com/hello/world/cmd/cmd1` (part of the `github.com/hello/world` module)
2. `github.com/hello/world/cmd/cmd2`, (also part of the `github.com/hello/world` module)
3. `acme.com/quote/cmd/quoter` (which belongs to the `acme.com/quote` module, which is a dependency of `github.com/hello/world`)
4. `github.com/random/module/cmd/hello` (part of the `github.com/random/module` module, which is _not_ a dependency of `github.com/hello/world`)

Is that a scenario `goreleaser` supports today? 

If it is, then to build on the point I was making in #1354, we need one temporary module for the builds of binaries 1 and 2. 

We then have a decision to make. Either for binary 3 we could use the same temporary module as for 1 and 2, which will ensure that the binary as built is as described by the dependency on the `acme.com/quote` module. Or alternatively we have a separate temporary module for 3 and another separate one for 4, because you're of the opinion that they should be built as those module authors intended. 

Would be good to understand whether this conversation is even relevant though - it might fall outside the scope of `goreleaser` in which case we only need to solve for binaries 1 and 2.

Reply:
in the current state of this pr, all those cases would work. Each would be a build definition, each will have its own temporary module folder (even if its the same module like 1 and 2).

But, I'm thinking we should only allow to build current module at current tag, otherwise, its a bit weird as I'm cloning one repo and building another completely different thing with no relation to it (as for example item 4 in your list).

On goreleaser stable, only cases 1 and 2 work, so I think we might stay with that 🤔

Reply:
Yep, per https://github.com/goreleaser/goreleaser/pull/2129#issuecomment-804867144 I think that's a sensible starting point.